examples/demos/create_infrastructure_with_storage.pp

```diff
@@ -0,0 +1,288 @@
+################################################################################
+# (C) Copyright 2020 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+# This example works for api_variant "Synergy"
+# Create FC Network
+oneview_fc_network{'Test Puppet FC Network':
+  ensure => 'present',
+  data   => {
+    name                    => 'Test Puppet FC Network',
+    connectionTemplateUri   => nil,
+    autoLoginRedistribution => true,
+    fabricType              => 'FabricAttach'
+    }
+}
+
+# Create Ethernet Network
+oneview_ethernet_network{'Test Puppet Network':
+  ensure => 'present',
+  data   => {
+    name           => 'Test Puppet Network',
+    vlanId         => '1045',
+    purpose        => 'General',
+    smartLink      => true,
+    privateNetwork => false
+  }
+}
+
+oneview_storage_system{'Test Puppet Storage System':
+    ensure => 'present',
+    data   => {
+      family   => 'StoreServ',
+      hostname => '172.18.11.11',
+      username => 'dcs',
+      password => 'dcs'
+    }
+}
+
+# Creates Storage System in OneView
+oneview_storage_pool{'Test Puppet Storage Pool':
+  ensure  => 'present',
+  require => Oneview_storage_system['Test Puppet Storage System'],
+  data    =>
+  {
+    poolName         => 'CPG-SSD-AO',
+    storageSystemUri => 'ThreePAR-1'
+  }
+}
+
+# Creates Storage Volume in OneView
+oneview_volume{'Test Puppet Storage Volume':
```

Comment:
why we have created this volume if we are not using this anywhere in the usecase

Reply:
I modified code in such a way SPT gets created with storage volume components.